### PR TITLE
Feature/move balls down + Snap to grid + Show Game-Over + small fixes

### LINKED ISSUE #3: Change coordinates for balls to float
The batch.draw() function takes floats as arguments for x and y, the Ball and BallManager use ints.

### PR DESCRIPTION
**Pull request by iiKoe**
Also fixes #3

### CODE REVIEW

**Comment by iiKoe** [comment]
I thought Jeroen would fix the to infinity part

File: core/src/com/group66/game/cannon/Ball.java [hunk 1]
```diff
@@ -199,12 +197,44 @@ public int getRadius() {
...
+	
+	/**
+	 * Sets the x coordinate.
+	 *
+	 * @param x the new x coordinate
+	 */
+	public void setX(float x) {
+		this.hitbox.x = x;
+		this.neighborBox.x = x;
+		this.topHitbox.x = x;
```
Comment (jeroenoverman) [inline core/src/com/group66/game/cannon/Ball.java]: Shouldn't it also change the x value of the ball itself?

File: core/src/com/group66/game/cannon/Ball.java [hunk 2]
```diff
@@ -199,12 +197,44 @@ public int getRadius() {
...
+	
+	/**
+	 * Sets the y coordinate.
+	 *
+	 * @param y the new y coordinate
+	 */
+	public void setY(float y) {
+		this.hitbox.y = y;
+		this.neighborBox.y = y;
+		this.topHitbox.y = y;
```
Comment (jeroenoverman) [inline core/src/com/group66/game/cannon/Ball.java]: Same as X

File: core/src/com/group66/game/cannon/Ball.java [hunk 3]
```diff
@@ -199,12 +197,44 @@ public int getRadius() {
...
 
 	/**
 	 * Update.
 	 * 
-	 * @param delta
-	 *            the change in time since the last update
+	 * @param delta the change in time since the last update
 	 */
 	public void update(float delta) {
 		hitbox.x += speed * (float) Math.cos(this.angle) * delta;
```
Comment (jeroenoverman) [inline core/src/com/group66/game/cannon/Ball.java]: also update neighbourbox and tophitbox?

File: core/src/com/group66/game/cannon/Ball.java (hunk 1 above)
Comment (iiKoe) [inline core/src/com/group66/game/cannon/Ball.java]: iirc the drawing is done according to the hitbox location

File: core/src/com/group66/game/cannon/Ball.java (hunk 3 above)
Comment (iiKoe) [inline core/src/com/group66/game/cannon/Ball.java]: This is for moving balls

**Pull request merged**